modules/social_features/social_group/modules/social_group_default_route/src/RedirectRoutesService.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace Drupal\social_group_default_route;
+
+use Drupal\Core\Extension\ModuleHandlerInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+
+/**
+ * Defines a redirect routes service class for manage group redirects.
+ */
+class RedirectRoutesService {
+
+  use StringTranslationTrait;
+
+  /**
+   * RedirectRoutesService constructor.
+   *
+   * @param \Drupal\Core\Extension\ModuleHandlerInterface $moduleHandler
+   *   The module handler.
+   */
+  public function __construct(
+    protected ModuleHandlerInterface $moduleHandler
+  ) {
+  }
+
+  /**
+   * Get default route for non-members.
+   *
+   * @return string
+   *   The default route.
+   */
+  public function getDefaultNonMemberRoute(): string {
+    // About.
+    return 'view.group_information.page_group_about';
+  }
+
+  /**
+   * Get default route for members.
+   *
+   * @return string
+   *   The default route.
+   */
+  public function getDefaultMemberRoute(): string {
+    // Stream.
+    return 'social_group.stream';
+  }
+
+  /**
+   * Get allowed routes for non-member.
+   *
+   * @return array
+   *   The array of routes.
+   */
+  public function getNonMemberRoutes(): array {
+    $routes = [
+      'view.group_information.page_group_about' => $this->t('About'),
+      'view.albums.page_group_albums_overview' => $this->t('Albums'),
+      'view.group_books.page_group_books' => $this->t('Books'),
+      'view.discussions.page_group_discussions' => $this->t('Discussions'),
+      'view.group_events.page_group_events' => $this->t('Events'),
+      'social_media_directory.group_files' => $this->t('Files'),
```

Comment:
@rochek03 this needs to moved to related modules.

Reply:
@ribel done, now new tabs can be added with `module_name.group_landing_tabs.yml` file with the next structure:
```
module_name.tab_name:
  title: Example
  route_name: example_route
  membership: member/non-member/all
  weight: 16
  // Isn't required, if it isn't set tabs will appear for all group types.
  group_types:
    - example_group_type
  // Isn't required, if it isn't set tabs will appear without conditions.
  conditions:
    group_field_name: value
```